src/Adapter/CartRule/Repository/CartRuleRepository.php

```diff
@@ -328,6 +350,27 @@ public function restrictCartRules(CartRuleId $cartRuleId, array $restrictedCartR
         );
     }
 
+    /**
+     * @param CartRuleId $cartRuleId
+     *
+     * @return int[]
+     */
+    public function getAssociatedShopIds(CartRuleId $cartRuleId): array
```

Comment:
this one concerns me as well.
Its almost the same as if we use CartRule->getAssociatedShops, expect it doesnt check Shop::isTableAssociated (which would return false and then empty array, because as mentioned in other comments, by default cart_rule_shop acts as "shop restriction" and is not a part of "shop association".
So it concerns me that those 2 methods will return different results. So it might cause alot of confusion if we don't fix it fully. (and we kind of cannot fix it fully until the old page is gone)

Reply:
I think indeed we should add `cart_rule` as a multishop table as well and then benefit from the existing multi shop feature from ObjectModel

But I'm not sure to understand why this would not be compatible with the legacy page. Did you already identify some stuff that would not work anymore or it's just an intuition?